src/OrchardCore/OrchardCore.ResourceManagement/TagHelpers/StyleTagHelper.cs

```diff
@@ -5,16 +5,21 @@
 
 namespace OrchardCore.ResourceManagement.TagHelpers
 {
+    [HtmlTargetElement("style", Attributes = IdAttributeName)]
     [HtmlTargetElement("style", Attributes = NameAttributeName)]
     [HtmlTargetElement("style", Attributes = SrcAttributeName)]
     [HtmlTargetElement("style", Attributes = AtAttributeName)]
     public class StyleTagHelper : TagHelper
     {
+        private const string IdAttributeName = "id";
```

Comment:
This should be `asp-id`.

It also shouldn't be part of the `HtmlTargetElement` attribute set (I think), as this would cause the tag helper to be activated when only using the `asp-id` attribute.

Activation should only happen on the name, src, or at attribute.

Reply:
I'm fine with asp-id. I did not create a lot of those TagHelpers I don't know all the tricks yet.